tests/server/amo/TestViews.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-loop-func */
+
+import request from 'supertest-as-promised';
+import { assert } from 'chai';
+
+import { runServer } from 'core/server/base';
+import Policy from 'csp-parse';
+
+import { checkSRI } from '../helpers';
+
+const defaultURL = '/en-US/firefox/';
+
+describe('AMO GET Requests', () => {
+  let app;
+
+  before(() => runServer({ listen: false, app: 'amo' })
+    .then((server) => {
+      app = server;
+    }));
+
+  after(() => {
+    webpackIsomorphicTools.undo();
+  });
+
+  it('should have a CSP policy on the amo app homepage', () => request(app)
+    .get(defaultURL)
+    .expect(200)
+    .then((res) => {
+      const policy = new Policy(res.header['content-security-policy']);
+      assert.notInclude(policy.get('script-src'), "'self'");
+      assert.include(policy.get('script-src'), 'https://addons.cdn.mozilla.net');
+      assert.notInclude(policy.get('connect-src'), "'self'");
+      assert.include(policy.get('connect-src'), 'https://addons.mozilla.org');
+    }));
+
+  it('should be using SRI for script and style on the amo app homepage', () => request(app)
+    .get(defaultURL)
+    .expect(200)
+    .then((res) => checkSRI(res)));
+
+  it('should be a 200 for requests to the homepage directly', () => request(app)
+    .get('/en-US/firefox/')
+    .expect(200));
+
+  it('should redirect an invalid locale', () => request(app)
+    .get('/whatevs/firefox/')
```

Comment:
I added these before I moved everything to a middleware since it was originally all in the base server and not easily testable outside of server tests.

As you say this does cover the integration of the middleware and the rest of the app but I should take a look at paring these down.